### DISCUSSION TITLE
PIP-5: address prefix

### DISCUSSION THREAD

**Discussion by ja7ad** [pactus-project/PIPs#16] [PIPs]
PIP-5: [address prefix](https://pips.pactus.org/PIPs/pip-5)

**Comment by b00f** [comment]
The Pactus address is made of 21 bytes. It includes the type of encryption used and the public key hash. You can learn more about it [here](https://pactus.org/learn/blockchain/address/).

Right now, the address doesn't show what it's used for. 
Because it doesn't show its use, people can make mistakes. For instance, during our Testnet-1, some users sent money to the wrong address. You can see an example [here](https://github.com/pactus-project/pactus/issues/510). Also, in our block explorer, users struggled to find validators because it always searched for accounts (bug in explorer).

I find this PIP (Proposal) an interesting idea. If we use it, the address can show its correct use. Like, we can stop a payment if it's going to the wrong type of address. Remember, the basic address stays the same (21 bytes), but how it looks can change based on its use.

## Wallet

Our wallet will also need an update. Currently, we set up addresses similar to this method [EIP-2334](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2334.md).

Our [current structure](https://github.com/pactus-project/pactus/blob/b56af2f40030618cdaf82dd72a27193f85538b58/wallet/vault/vault.go#L13) for creating addresses is:

```
m / purpose' / coin_type' / account / use
```

For now, the "use" section is always zero. We can change it to:
```
0: Account
1: Validator
```
So, when someone makes a new address, they need to pick its use.

**Comment by b00f** [comment]
The issue with this PIP is that the address doesn't communicate its type. Consequently, any given 21-byte sequence could represent both a validator and an account.
